### PR TITLE
fix type issues and package installs #48

### LINKED ISSUE #48: Issue with types missing, possibly incorrectly published package???
I'm getting back to my project after several months and I'm bogged down with a couple of issues from `nestjs-query-typegoose` and to be honest, I'm not sure it's just my incompetence or what. 

At any rate, this error is happening while building/ compiling: 

src/user/user.module.ts(2,44): error TS7016: Could not find a declaration file for module '@ptc-org/nestjs-query-typegoose'. '/home/scott/Documents/tmp/m8a-new/common/temp/node_modules/.pnpm/@ptc-org+nestjs-query-typegoose@1.1.0_aefggeysizmixjxhkqtfgu72mu/node_modules/@ptc-org/nestjs-query-typegoose/src/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/ptc-org__nestjs-query-typegoose` if it exists or add a new declaration (.d.ts) file containing `declare module '@ptc-org/nestjs-query-typegoose';`

This happens when I have `strict: true` set. If I set it to "off", the build goes through with no errors, but then I get this error while running my app:

Error: Cannot find module './module'
Require stack:
- /home/scott/Documents/tmp/m8a-new/common/temp/node_modules/.pnpm/@ptc-org+nestjs-query-typegoose@1.1.0_aefggeysizmixjxhkqtfgu72mu/node_modules/@ptc-org/nestjs-query-typegoose/src/index.js
- /home/scott/Documents/tmp/m8a-new/packages/core/dist/user/user.module.js
- /home/scott/Documents/tmp/m8a-new/packages/core/dist/auth/auth.module.js
- /home/scott/Documents/tmp/m8a-new/packages/core/dist/auth/index.js
- /home/scott/Documents/tmp/m8a-new/packages/core/dist/index.js
- /home/scott/Documents/tmp/m8a-new/packages/starter-kit-api/dist/app.module.js
- /home/scott/Documents/tmp/m8a-new/packages/starter-kit-api/dist/main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:955:15)
    at Function.Module._load (node:internal/modules/cjs/loader:803:27)
    at Module.require (node:internal/modules/cjs/loader:1021:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/home/scott/Documents/tmp/m8a-new/common/temp/node_modules/.pnpm/@ptc-org+nestjs-query-typegoose@1.1.0_aefggeysizmixjxhkqtfgu72mu/node_modules/@ptc-org/nestjs-query-typegoose/src/index.js:4:16)
    at Module._compile (node:internal/modules/cjs/loader:1119:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1173:10)
    at Module.load (node:internal/modules/cjs/loader:997:32)
    at Function.Module._load (node:internal/modules/cjs/loader:838:12)
    at Module.require (node:internal/modules/cjs/loader:1021:19)

It's like the package wasn't built properly to begin with??? 

### Have you read the [Contributing Guidelines](https://github.com/tripss/nestjs-query/blob/master/CONTRIBUTING.md)?

Yes

**To Reproduce**

Don't have any reproduction repo. Sorry. If it is a prerequisite, let me know. 

**Expected behavior**
I'm expecting none of these errors. 

**Desktop (please complete the following information):**
 - Node Version: 18.9.0
 - Nestjs-query Version: 1.1.0

I hope you can help me on this, as it's blocking any further work. Thanks!

Scott

### PR DESCRIPTION
**Pull request by smolinari**
Hope this is ok and it works to fix #48. 

Scott

### CODE REVIEW

File: packages/query-typegoose/package.json [hunk 1]
```diff
@@ -25,10 +25,10 @@
     "tslib": "^2.4.0"
   },
   "peerDependencies": {
+    "@m8a/nestjs-typegoose": "^8.0.5",
```
Comment (TriPSs) [inline packages/query-typegoose/package.json:28]: What is this one?

Comment (smolinari) [reply packages/query-typegoose/package.json:28]: It's the package I've forked and am maintaining for including Typegoose into Nest. Similar to how you are taking over for nestjs-query. :grinning: 

Scott

Comment (smolinari) [reply packages/query-typegoose/package.json:28]: What breaks? I can look into that. 

Scott

Comment (smolinari) [reply packages/query-typegoose/package.json:28]: Ah. Wait. I think I know why. But that is strange. I didn't see any failing tests. Hmm.... let me have a look. 

Scott

Comment (smolinari) [reply packages/query-typegoose/package.json:28]: Ok. I see the problem. I have to update the top-level package.json too. Oh brother. Now I remember why I dislike Nx. :shrug: 

Scott

Comment (TriPSs) [reply packages/query-typegoose/package.json:28]: Well the code still uses the original package, NX can't handle/does nothing with peer deps as normal dev deps are added when building the project.

Could we to the migration to that one in separate PR?

Comment (smolinari) [reply packages/query-typegoose/package.json:28]: Unfortunately, that is one of the main culprits for seriously lacking in upgrading. I'm also almost done. I've also had to up the TypeScript version to the newest (might as well, right?) and I'm fixing the issues that has caused. The newest Mongoose types were breaking with 4.4.4. :shrug: 

Scott

**Pull request closed without merging**